### PR TITLE
Gray suit and variants

### PR DESCRIPTION
**Pull request by Asddsa76**
New suit: Gray, touched by no colours and critical

New variants: Gray (5 suits), Gray (6 suits), Black and Gray (6 suits), Gray and Rainbow (6 Suits), Gray and Dark Rainbow (6 Suits).

Proposed Triple Black variant (Grey, Black, and Dark Rainbow (6 suits)) scrapped.

Gray, the suit for rank clued cards of unknown colour, changed to Gray_Unknown.

### CODE REVIEW

**Comment by Zamiell** [comment]
squash all your commits into 1 commit and then i'll take a look

File: docs/VARIANTS.md [hunk 1]
```diff
@@ -48,6 +48,12 @@ Hanabi Live is programmed by Hanabi enthusiasts who have played thousands of gam
 * All color clues will "touch" the dark rainbow suit.
 * There is only one of each dark rainbow card in the deck, which means that every dark rainbow card is "critical".
 
+### Gray
+
+* One of the suits is replaced with a gray suit.
+* No color clues "touch" the gray suit. (It is a colorless suit.)
```
Comment (rogers-j) [inline docs/VARIANTS.md]: This is just like white suit.....

File: public/js/src/constants.js [hunk 2]
```diff
@@ -312,6 +313,14 @@ exports.SUIT = {
         [exports.COLOR.BLACK],
         true, // This suit has one of each card
     ),
+    GREY: new Suit(
```
Comment (rogers-j) [inline public/js/src/constants.js]: Yes, yes, having both GREY and GRAY is very clever.

Too clever, even.

File: docs/VARIANTS.md (hunk 1 above)
Comment (Asddsa76) [reply docs/VARIANTS.md]: Line 55 says it's critical

**Comment by Asddsa76** [comment]
I squashed them

File: docs/VARIANTS.md [hunk 3]
```diff
@@ -109,8 +115,6 @@ Hanabi Live is programmed by Hanabi enthusiasts who have played thousands of gam
   5. Rainbow (all colors)
   6. Black (one of each)
 
-<br />
```
Comment (Zamiell) [inline docs/VARIANTS.md]: why did you remove this?

File: public/js/src/constants.js [hunk 4]
```diff
@@ -280,10 +281,10 @@ exports.SUIT = {
     ),
 
     // This represents cards of unknown suit; it must not be included in variants
-    GRAY: new Suit(
-        'Gray',
+    GRAY_UNKNOWN: new Suit(
+        'GRAY_UNKNOWN',
```
Comment (Zamiell) [inline public/js/src/constants.js]: Change this to just "Gray'

File: public/js/src/constants.js [hunk 5]
```diff
@@ -818,6 +827,67 @@ exports.VARIANTS = {
...
+        false,
+    ),
+    'Gray and Dark Rainbow (6 Suits)': new Variant(
+        [
+            exports.SUIT.BLUE,
+            exports.SUIT.GREEN,
+            exports.SUIT.YELLOW,
+            exports.SUIT.RED,
+            exports.SUIT.GRAY,
+            exports.SUIT.DARKRAINBOW,
```
Comment (Zamiell) [inline public/js/src/constants.js]: change this to DARK_RAINBOW

File: public/js/src/game/HanabiCard.js [hunk 6]
```diff
@@ -48,7 +48,7 @@ class HanabiCard extends Phaser.GameObjects.Container {
         this.add(image);
 
         /*
-        // Create the "bare" card image, which is a gray card with all the pips
+        // Create the "bare" card image, which is a GRAY_UNKNOWN card with all the pips
```
Comment (Zamiell) [inline public/js/src/game/HanabiCard.js]: wtf?

File: public/js/src/game/HanabiCard.js [hunk 7]
```diff
@@ -126,13 +126,13 @@ class HanabiCard extends Phaser.GameObjects.Container {
...
-        let suitToShow = suit || learnedCard.suit || SUIT.GRAY;
+        let suitToShow = suit || learnedCard.suit || SUIT.GRAY_UNKNOWN;
         if (empathyPastSuitUncertain) {
-            suitToShow = SUIT.GRAY;
+            suitToShow = SUIT.GRAY_UNKNOWN;
         }
 
-        // "Card-Gray" is not created, so use "NoPip-Gray"
-        if (suitToShow === SUIT.GRAY) {
+        // "Card-GRAY_UNKNOWN" is not created, so use "NoPip-GRAY_UNKNOWN"
```
Comment (Zamiell) [inline public/js/src/game/HanabiCard.js]: you should look at your commits before submitting a pull request and you would see errors like this

File: public/js/src/game/drawCards.js [hunk 8]
```diff
@@ -95,23 +95,23 @@ exports.drawAll = () => {
             // 'NoPip' cards are used for
             //   cards of known rank before suit learned
             //   cards of unknown rank
-            // Entirely unknown cards (Gray 6) have a custom image defined separately
-            if (rank > 0 && (rank < 6 || suit !== SUIT.GRAY)) {
+            // Entirely unknown cards (GRAY_UNKNOWN 6) have a custom image defined separately
```
Comment (Zamiell) [inline public/js/src/game/drawCards.js]: fix this

File: public/js/src/game/drawCards.js [hunk 9]
```diff
@@ -95,23 +95,23 @@ exports.drawAll = () => {
...
-            if (suit !== SUIT.GRAY) {
+            // GRAY_UNKNOWN Card images would be identical to NoPip images
+            if (suit !== SUIT.GRAY_UNKNOWN) {
                 globals.ui.cardImages[`Card-${suit.name}-${rank}`] = cvs;
             }
         }
     }
 
-    globals.ui.cardImages['NoPip-Gray-6'] = makeUnknownCardImage();
+    globals.ui.cardImages['NoPip-GRAY_UNKNOWN-6'] = makeUnknownCardImage();
```
Comment (Zamiell) [inline public/js/src/game/drawCards.js]: you fucked up

File: public/js/src/game/ui/HanabiCard.js [hunk 10]
```diff
@@ -117,34 +117,34 @@ class HanabiCard extends graphics.Group {
...
             && !learnedCard.revealed
             && !this.isClued()
             && (this.holder === globals.playerUs || this.empathy)
         ) {
-            suitToShow = constants.SUIT.GRAY;
+            suitToShow = constants.SUIT.GRAY_UNKNOWN;
         }
 
-        // For whatever reason, "Card-Gray" is never created, so use "NoPip-Gray"
+        // For whatever reason, "Card-GRAY_UNKNOWN" is never created, so use "NoPip-GRAY_UNKNOWN"
```
Comment (Zamiell) [inline public/js/src/game/ui/HanabiCard.js]: you fucked up

File: public/js/src/game/ui/drawCards.js [hunk 11]
```diff
@@ -95,23 +95,23 @@ exports.drawAll = () => {
             // 'NoPip' cards are used for
             //   cards of known rank before suit learned
             //   cards of unknown rank
-            // Entirely unknown cards (Gray 6) have a custom image defined separately
-            if (rank > 0 && (rank < 6 || suit !== SUIT.GRAY)) {
+            // Entirely unknown cards (GRAY_UNKNOWN 6) have a custom image defined separately
```
Comment (Zamiell) [inline public/js/src/game/ui/drawCards.js]: wrong

File: public/js/src/game/ui/drawCards.js [hunk 12]
```diff
@@ -95,23 +95,23 @@ exports.drawAll = () => {
...
             }
 
-            if (suit !== SUIT.GRAY) {
+            if (suit !== SUIT.GRAY_UNKNOWN) {
                 drawSuitPips(ctx, rank, suit, i);
             }
 
-            // Gray Card images would be identical to NoPip images
-            if (suit !== SUIT.GRAY) {
+            // GRAY_UNKNOWN Card images would be identical to NoPip images
```
Comment (Zamiell) [inline public/js/src/game/ui/drawCards.js]: wrong

File: public/js/src/game/ui/drawCards.js [hunk 13]
```diff
@@ -95,23 +95,23 @@ exports.drawAll = () => {
...
+            if (suit !== SUIT.GRAY_UNKNOWN) {
                 globals.cardImages[`Card-${suit.name}-${rank}`] = cvs;
             }
         }
     }
 
-    globals.cardImages['NoPip-Gray-6'] = makeUnknownCardImage();
+    globals.cardImages['NoPip-GRAY_UNKNOWN-6'] = makeUnknownCardImage();
```
Comment (Zamiell) [inline public/js/src/game/ui/drawCards.js]: wrong

File: public/js/src/game/ui/drawReplayArea.js [hunk 14]
```diff
@@ -71,7 +71,7 @@ module.exports = () => {
         width: shuttleValues.w * winW,
         height: shuttleValues.h * winH,
         cornerRadius: shuttleValues.cornerRadius * winW,
-        fill: '#d1d1d1', // Gray
+        fill: '#d1d1d1', // GRAY_UNKNOWN
```
Comment (Zamiell) [inline public/js/src/game/ui/drawReplayArea.js]: wrong

File: public/js/src/constants.js (hunk 4 above)
Comment (rogers-j) [reply public/js/src/constants.js]: Or just `UNKNOWN`?

Comment (Zamiell) [reply public/js/src/constants.js]: yeah, UNKNOWN is a better name IMO
